### PR TITLE
feat: add a homey-app config

### PR DESCRIPTION
**Pull request by tjallingt**
Adds a new config for Homey Apps that can be extended from like this: 

```json
{ "extends": "athom/homey-app" }
```

This config contains a new plugin called `eslint-plugin-homey-app` which warns for usage of `setTimeout()` and `setInterval()`.

There are also some changes to the base config. I've moved from `babel-eslint` to `@typescript-eslint/parser`, both are able to parse modern JavaScript and TypeScript so I believe this change does not affect ESLint behaviour.
The reason to use `@typescript-eslint/parser` is so we can use the `@typescript-eslint/eslint-plugin`. Due to type checking this plugin has rules like `no-floating-promises` which detects calls that are missing `await`/`.catch()`.

This does require the presence of a `tsconfig.json` in the root of the project, so Homey Apps that want to use the `athom/homey-app` config need to add that file.

We can probably add some sort of migration to the Homey CLI to upgrade an app.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -15,20 +17,24 @@
...
   "dependencies": {
-    "eslint-config-airbnb-base": "^14.0.0",
-    "eslint-plugin-import": "^2.20.0",
-    "eslint-plugin-node": "^11.0.0",
-    "eslint-plugin-mocha": "^6.2.2",
-    "babel-eslint": "^10.0.3"
+    "@typescript-eslint/eslint-plugin": "^4.31.1",
+    "@typescript-eslint/parser": "^4.31.1",
+    "eslint-config-airbnb-base": "^14.2.1",
+    "eslint-plugin-homey-app": "github:athombv/eslint-plugin-homey-app",
```
Comment (tjallingt) [inline package.json:23]: TODO: this plugin should be published

**Review by RobinBol — APPROVED** [review]
Maybe we should doc the required `.tsconfig` somewhere at least?

File: README.md [hunk 2]
```diff
@@ -24,3 +24,26 @@ Now, edit your project's `/package.json` file to contain the following:
...
+
+This package also contains a ESLint config for Homey Apps, you can extend it like this:
+
+```javascript
+{
+  "extends": "athom/homey-app"
+}
+```
+
+This config includes some rules that depend on TypeScript typechecking so you need to make sure to have a `tsconfig.json` file in the root of your Homey App.
```
Comment (RobinBol) [inline README.md:38]: Are there any requirements as to the contents of `tsconfig.json` the developer should know about?

Comment (tjallingt) [reply README.md:38]: Mhh not really although it should be usable by typescript. Will improve these instructions when I add a command to the Homey CLI to automate this.

**Pull request merged**